### DISCUSSION TITLE
Question about abilityfeatures file

### DISCUSSION THREAD

**Discussion by Teamtorrentscripts** [Albeoris/Memoria#460] [Q&A] [answered]
Hello, I am seeing that the abilityfeatures file contains many characters and scripts to modify or create interesting abilities, the question is, can I somehow make an enemy give me a specific object by killing him using this? without having to resort to editing the enemies' own rewards with HW

Would something like this example be used to get the item with ID 38 when winning the battle?

BattleResult [code=WeaponId] 38 [/code]

I appreciate whoever helps me.

**Comment by Tirlititi** [comment, marked as answer]
Hello. Yes, that is possible, although I'm not 100% sure I understood correctly what you want to do.
If it's about "this enemy gives that reward but only when a character has that weapon equipped", then it would be done with something like that:
```txt
>SA Global+
BattleResult WhenRewardAll
[code=Condition] WeaponId == RegularItem_HolyLance && BattleId == 525 [/code]
[code=BonusItem1] RegularItem_DragonWrist [/code]
[code=BonusItemCount1] 1 [/code]
```
That way, you get a Dragon Wrist as a reward for fighting Valia Pira when someone is equipped with the Holy Lance.
The battle IDs can be found [here.](https://github.com/Albeoris/Memoria/wiki/List-of-battles)

EDIT: with the latest commits, using these lines instead don't replace the original first reward by the Dragon Wrist:
```txt
>SA Global+
BattleResult WhenRewardAll
[code=Condition] WeaponId == RegularItem_HolyLance && BattleId == 525 [/code]
[code=BonusItemAdd] RegularItem_DragonWrist ; 1 [/code]
```

**Reply by Teamtorrentscripts** [reply]
Yeah! That's what I was referring to, but that's so you always get the item at the end of the fight, right? Isn't there something to make it random with "GetRandom"?

Thank you so much.

**Reply by Tirlititi** [reply]
Sure, you add a part in the condition for that:
```txt
[code=Condition] WeaponId == RegularItem_HolyLance && BattleId == 525 && GetRandom(0, 100) < 33 [/code]
```
Here, you must (1) have the Holy Lance equipped, (2) be fighting against Valia Pira and then (3) you'll have 33% chance of getting that reward.

I think you'll be replacing a reward (if there was) when doing that. Maybe using ``BonusItem6`` and ``BonusItemCount6`` avoids replacing any reward in almost all situations, but maybe that will add unwanted visuals... I'll add a better way to handle "add to the rewards".

**Reply by Teamtorrentscripts** [reply]
I understand, I will try what you tell me, thank you very much once again!!

**Comment by Teamtorrentscripts** [comment]
Another question regarding Ability features, how could we add the way in which when activating an SA ability, the target of the spells can be changed? For example, when activating SA0, the attack command is for multiple enemies.

**Reply by Tirlititi** [reply]
You would do an ability patch for that, with the base ability being single-target and the improved ability being multi-target. Say they have IDs 100 and 101 respectively.
```txt
>AA 100
[code=Patch] HasSA(0) ? 101 : -1 [/code]
```
This can be used for any kind of "improved version of an ability", not only changing from single to multi-target. It allows to have a different description for the improved version and will be taken into account out of battles as well.

**Reply by Teamtorrentscripts** [reply]
Thank you very much, I've been trying everything you told me for a while and it works perfectly.

Now how can I add elemental effects to weapons using SA? For example, when activating SA0 it is a fire element and takes more life from an enemy who is vulnerable to said element.
